wallet/wallet-cli-lib/src/commands/mod.rs

```diff
@@ -657,14 +663,21 @@ impl CommandHandler {
                 }
             }
 
-            WalletCommand::GenerateBlock { transactions } => {
+            WalletCommand::GenerateBlock {
+                pool_id,
+                transactions,
```

Comment:
Oh, my... this brings even another level of complexity to the equation. Maybe @alfiedotwtf and @iljakuklic can chime in after the long discussions that were had about adding transactions manually with generate_block in Alfie's PR. What happens now when multiple pools take the same transactions and try to put them in blocks? Will it always work without issues and without double-spends?

Reply:
I don't see a problem here because only one pool can succeed after the `generate_block` call.